firmware/t2t1/releases.json

```diff
@@ -166,6 +177,7 @@
         "url_bitcoinonly": "data/firmware/t2t1/trezor-t2t1-2.4.0-bitcoinonly.bin",
         "fingerprint": "d90265ee6d7d499c7d938b5322f71f27042da8a6fdaed54c224d31b65e868def",
         "fingerprint_bitcoinonly": "89c91287ab7a9cd3ec246b6822a0d04b7d40401abef706cccafbb7b98bd6a3d7",
+        "firmware_revision": "656133353936616438",
```

Comment:
From this line and below all revisions are broken. It almost seems like intentional.

Can we have a script to quickly check the revisions against tags? Should not be that hard to write

just call `git tag --points-at $COMMIT` for the provided revision and check if it is sane.

Reply:
I’ve used the data provided by Andrew from [here](https://satoshilabs.slack.com/archives/CL1D61PQF/p1721833314507079?thread_ts=1721826646.998999&cid=CL1D61PQF)

Reply:
@peter-sanderson can you help me with this please?

Reply:
Ah, I see. There was a bug in some older versions of the firmware and they returned bytes instead of string.

For example `623139636266363763` is hex encoding of `b19cbf67c` which is the real revision.

`git log 623139636266363763` returns nothing while `git log b19cbf67c` returns commit with tag `core/v2.3.6`

Not sure what is the purpose of this field in JSON, but if it is supposed to store full revision there should be `b19cbf67c6c7c38513947b703df6d4409c59bc98`. If it is supposed to store revision returned by trezor, it should indeed store `623139636266363763` but in that case naming the field `firmware_revision` is confusing.

Reply:
We need those revision for implementation of [this](https://github.com/trezor/trezor-suite-private/issues/104)

Reply:
Where will the revision be read from? Will it be from the device, and will the suite compare it against this list?

Reply:
> Where will the revision be read from? Will it be from the device, and will the suite compare it against this list?

Yes. If Suite finds it for given firmware version locally, it will use local `releases.json` file. If not, it will try to find it online.